src/environmentd/tests/sql.rs

```diff
@@ -2309,3 +2309,80 @@ fn test_emit_tracing_notice() {
         x => panic!("failed to read message from channel, {:?}", x),
     }
 }
+
+#[test]
+fn test_subscribe_on_dropped_source() {
+    fn test_subscribe_on_dropped_source_inner(tables: Vec<&'static str>, subscribe: &'static str) {
+        let config = util::Config::default();
+        let server = util::start_server(config).unwrap();
+
+        server.runtime.block_on(async {
+            let (client, _conn_task) = server.connect_async(tokio_postgres::NoTls).await.unwrap();
+            for table in &tables {
+                client
+                    .batch_execute(&format!("CREATE TABLE {table} (a INT);"))
+                    .await
+                    .unwrap();
+                client
+                    .batch_execute(&format!("INSERT INTO {table} VALUES (1);"))
+                    .await
+                    .unwrap();
+            }
+
+            let (subscribe_client, _subscribe_conn_task) =
+                server.connect_async(tokio_postgres::NoTls).await.unwrap();
+            let subscribe_task = task::spawn(|| "subscribe", async move {
+                subscribe_client
+                    .batch_execute(&format!("SUBSCRIBE {subscribe}"))
+                    .await
+                    .unwrap();
+            });
+
+            Retry::default()
+                .max_duration(Duration::from_secs(10))
+                .retry_async(|_| async {
+                    let row = client
+                        .query_one("SELECT count(*) FROM mz_internal.mz_compute_exports;", &[])
+                        .await
+                        .unwrap();
+                    let count: i64 = row.get(0);
+                    if count < 1 {
+                        Err("no active subscribe")
+                    } else {
+                        Ok(())
+                    }
+                })
+                .await
+                .unwrap();
+
+            for table in &tables {
+                client
+                    .batch_execute(&format!("DROP TABLE {table};"))
```

Comment:
This PR looks like it drops subscribes while they are pending. Have the subscribes already exited pending by the time we're running the DROP? That is: what does this test do without the code changes?

Reply:
Without the code changes, this test hangs forever because the subscribe is never dropped.

Reply:
I think pending is a bit of a misleading term used in the Coordinator, and a better name would have been `active_subscribes`. Once a subscribe enters the `pending_subscribes` map, then a sink has been created for it and it will serve results. The subscribe doesn't leave the `pending_subscribes` map until the subscribe has been cancelled or committed.

Reply:
Ah, I thought it was like peeks. Ok.

Reply:
I'm honestly considering changing the name to `active_subscribes` as part of this PR. Would you be on board for that?

Reply:
Yes.